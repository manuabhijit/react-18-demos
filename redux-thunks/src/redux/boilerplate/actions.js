const addNotesAction = (myNoteObject) => ({
  type: 'ADD_NOTE',
  myNoteObject,
});

export default addNotesAction;
