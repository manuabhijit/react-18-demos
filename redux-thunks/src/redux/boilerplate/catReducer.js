// Todo: separate out cat reducer