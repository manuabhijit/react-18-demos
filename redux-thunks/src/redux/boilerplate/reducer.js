export const initialState = {
  uiNotes: [
    {
      title: "First Note",
    },
    {
      title: "Second Note",
    },
  ],
  catFact: [],
  counter: 1,
};

const notesReducer = (state = initialState, action = {}) => {
  console.log(action);
  switch (action.type) {
    case "ADD_NOTE": {
      const uiNotes = state.uiNotes;
      uiNotes.push(action.myNoteObject);

      state = {
        ...state,
        uiNotes: JSON.parse(JSON.stringify(uiNotes)),
      };
      break;
    }
    case "ADD_CAT_FACT": {
      console.log(action);
      state = {
        ...state,
      }
      break;
    }
    default: {
      console.log("UNKNOWN ACTION");
    }
  }

  return state;
};

export default notesReducer;
