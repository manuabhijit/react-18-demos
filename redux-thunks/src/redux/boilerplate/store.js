import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import notesReducer from './reducer';
import thunk from 'redux-thunk';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const notesStore = createStore(
  combineReducers({
    uiNotes: notesReducer,
  }),
  composeEnhancers(
    applyMiddleware(
      // logger
      thunk
    ),
  ),
);

export default notesStore;
