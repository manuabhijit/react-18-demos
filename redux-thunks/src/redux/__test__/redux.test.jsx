import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState } from "../boilerplate/reducer";
import ReduxComponent from "../redux";


describe("Test: Redux component screenshot", () => {
  const middleWares = [thunk];
  const mockStore = configureStore(middleWares);

  it("should render correctly", () => {
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <ReduxComponent />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
