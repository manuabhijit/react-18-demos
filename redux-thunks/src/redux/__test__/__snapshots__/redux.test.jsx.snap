// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Special Mention component screenshot should render the Special Mention correctly 1`] = `
<div>
  <div>
    <h1>
      REDUX demo
    </h1>
    <br />
    <input />
    <button>
      Add
    </button>
  </div>
</div>
`;
